numpyro/contrib/tfp/distributions.py

```diff
@@ -24,7 +24,7 @@ def _get_codomain(bijector):
         return constraints.positive
     elif bijector.__class__.__name__ == "GeneralizedPareto":
         loc, scale, concentration = bijector.loc, bijector.scale, bijector.concentration
-        if not_jax_tracer(concentration) and np.all(concentration < 0):
+        if not_jax_tracer(concentration) and np.all(np.less(concentration, 0)):
```

Comment:
Currently,
```
import jax
from numpyro.util import not_jax_tracer
import numpy as np
import jax.numpy as jnp

def f(x):
    if not_jax_tracer(y):
        if np.all(y <= 0):
            print("nooo")
    return x + y

y = jnp.arange(10.)
jax.jit(f)(1)
```
will fail because although `concentration` is not a tracer, `concentration < 0` will be a tracer if concentration is a DeviceArray. For validation checks like this, we need to use all numpy ops.

Reply:
These nuances are getting very subtle as compared to the earlier behavior, and I am not sure if I fully understand when to use `numpy` vs `jax.numpy`, and now there is this additional distinction between numpy and python ops. :)

Reply:
Yeah, for device array, python op `concentration < 0` will be deferred to something like `jnp.less(concentration, 0)` or so. The rule of thumb now is to always use `numpy` ops (not python ops). It is quite unfortunate. >___<

Reply:
Thanks for explaining! Any idea what's the larger motivation for this change? Are we moving towards a world where everything will be JITed by default?

Reply:
I am not sure. This behavior has appeared since jax 0.2. I haven't caught the fact that `a < 0` will trigger a jax op if a is a DeviceArray until now.